### PR TITLE
Make the build green 

### PR DESCRIPTION
**Pull request by bonomat**
* Remove MacOS e2e test
* Cleared target cache as it seemed to have a problem there
* Removed the GitHub workflow file linter and added it as a pre-commit action

Note: the linter has a problem currently as tracked in this ticket: https://github.com/stoplightio/spectral/issues/1097
That means, if you edit the workflow files and try to commit, you will most likely get an error.

### CODE REVIEW

File: .github/workflows/lint-workflow-files.yml [hunk 1]
```diff
@@ -13,4 +13,4 @@ jobs:
         uses: actions/setup-node@v1
 
       - name: Run spectral
-        run: npx @stoplight/spectral lint .github/workflows/*.yml --ignore-unknown-format --verbose
+        run: npx @stoplight/spectral lint [.github/workflows/*.yml] --ignore-unknown-format --verbose
```
Comment (D4nte) [inline .github/workflows/lint-workflow-files.yml:16]: 📣 This prevent the check to happen.

Comment (D4nte) [reply .github/workflows/lint-workflow-files.yml:16]: I may be wrong, please confirm it still checks and if so, please freeze spectral's version.

Comment (D4nte) [reply .github/workflows/lint-workflow-files.yml:16]: Actually I'd question the use of this github action. Move it to a git pre commit hook instead...

Comment (bonomat) [reply .github/workflows/lint-workflow-files.yml]: I agree with you. It bugged me more than once already. 
I've removed it from GitHub CI and added it to the pre-commit action.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -78,27 +78,23 @@ jobs:
...
-          name: cnd-${{ matrix.os }}
+          name: cnd
           path: target/debug/cnd
 
   e2e_test:
-    strategy:
-      # Do not cancel ubuntu even if macos fails
-      fail-fast: false
-      matrix:
-        os: [macos-latest, ubuntu-latest]
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml:91]: Could have also just removed `macos-latest` here, no?

Comment (D4nte) [reply .github/workflows/ci.yml:91]: @bonomat please note I privileged this method to keep the nice `(ubuntu-latest)` after the GitHub check name to not have to modify all bors/mergify/github branch protection configurations.

**Comment by D4nte** [comment]
Bors r+

File: .github/workflows/ci.yml (hunk 2 above)
Comment (bonomat) [reply .github/workflows/ci.yml:91]: @thomaseizinger 
> Could have also just removed macos-latest here, no?

probably :D

**Pull request merged**